addons/mail/static/src/core/web/activity.js

```diff
@@ -128,4 +129,16 @@ export class Activity extends Component {
     get thread() {
         return this.threadService.getThread(this.props.data.res_model, this.props.data.res_id);
     }
+
+    /**
+     * @param {MouseEvent} ev
+     */
+    async onClick(ev) {
+        let partnerId = null;
+        if (this.props.data.user_id) {
+            const user = await this.env.services.orm.read("res.users", [this.props.data.user_id[0]], ["partner_id"]);
+            partnerId = user?.[0]?.partner_id?.[0];
+        }
```

Comment:
This is not the correct partner, I think it just aims to be `store.self`, but as posted elsewhere you can remove this.